account-model/src/account/account_builder.rs

```diff
@@ -0,0 +1,321 @@
+            //     return validatePtAccountRestriction(&self.account);
+            // } else if self.account.country == CountryCode::ESP {
+            //     self.account.currency = Currency::EUR;
+            //     return validateEsAccountRestriction(&self.account);
+            // } else if self.account.country == CountryCode::CHE {
+            //     self.account.currency = Currency::CHF;
+            //     return validateChAccountRestriction(&self.account);
+            // } else if self.account.country == CountryCode::USA {
+            //     self.account.currency = Currency::USD;
+            //     return validateUsAccountRestriction(&self.account);
+        } else {
+            Err(format!("Unsupported country {}", self.country.unwrap()))
+        }
+    }
+
+    fn create_account_with_gb_country_restrictions(&mut self) -> Result<Account, String> {
+        if self.bic.is_none() {
+            return Err(format!("{} requires Bic", CountryCode::GBR.alpha3()));
+        }
+        self.validate_account_restriction_by(CountryCode::GBR,
+                                             8,
+                                             6,
+                                             KnownBankIdCode::GbBankIdCode)?;
+        Ok(Account {
+            id: get_or_create_id(self.id),
+            organisation_id: get_or_create_id(self.organisation_id),
+            country: CountryCode::GBR,
+            currency: Currency::GBP,
+            bank_id: self.bank_id.as_ref().expect("bank id is required").to_string(),
+            bank_id_code: self.bank_id_code.as_ref().expect("bank id code is required").to_string(),
+            bic: self.bic.as_ref().unwrap().to_string(),
+            number: self.number.clone(),
+            iban: self.iban.clone(),
+            title: self.title.clone(),
+            classification: self.classification,
+        })
+    }
+
+    fn validate_account_restriction_by(&mut self,
+                                       country: CountryCode,
+                                       expected_number_length: u8,
+                                       expected_bank_id_length: u8,
+                                       expected_code: KnownBankIdCode) -> Result<(), String> {
+        if self.number.is_some() {
+            if self.number.as_ref().unwrap().len() != expected_number_length as usize {
+                return Err(format!("{} requires {}-character long Account Number",
+                                   country.alpha3(), expected_number_length));
+            }
+        }
+        if self.bank_id.is_none() || self.bank_id.as_ref().unwrap().len() != expected_bank_id_length as usize {
+            return Err(format!("{} requires {}-character BankId, got '{:?}'",
+                               country.alpha3(), expected_bank_id_length, self.bank_id));
+        }
+
+        let bank_id_code: &str = expected_code.into();
+        let bank_id_code_s: &String = &bank_id_code.to_string();
```

Comment:
czy mogę to jakoś innaczej zapisać ?

Reply:
nie rozumiem pytania

Reply:
into nie wie na co przekonwertowac wiec musze zadeklarować najpierw `bank_id_code: &str` a potem to przekonwertować na `&String` tylko poto zeby przyrównać linijke niżej, da się to może zrobić w jenej lini?